src/Traveler.js

```diff
@@ -0,0 +1,91 @@
+import * as dayjs from 'dayjs';
+var isBetween = require('dayjs/plugin/isBetween');
+dayjs.extend(isBetween);
+
+class Traveler {
+  constructor(travelerDetails, trips) {
+    this.id = travelerDetails.id;
+    this.name = travelerDetails.name;
+    this.travelerType = travelerDetails.travelerType;
+    this.allTrips = trips;
+    this.pendingTrips = [];
+    this.currentTrip = {};
+    this.pastTrips = [];
+    this.upcomingTrips = [];
+    this.spendingYTD = 0;
+  }
+
+  // maybe put this functionaility in scripts when getting the individual users trips
```

Comment:
Still considering this. Think about what class the functionality really belongs to...